connector/src/rest/connector/libs/dcnm/implementation.py

```diff
@@ -0,0 +1,370 @@
+__author__ = "Sukanya Kalluri <sukkallu@cisco.com>"
+
+import json
+import logging
+import requests
+from requests.auth import HTTPBasicAuth
+from requests.exceptions import RequestException
+
+from pyats.connections import BaseConnection
+from rest.connector.implementation import Implementation
+from rest.connector.utils import get_username_password
+
+# create a logger for this module
+log = logging.getLogger(__name__)
+STD_HEADER = {'Content-type': 'application/json'}
+
+
+class Implementation(Implementation):
+    '''Rest Implementation for DCNM
+
+    Implementation of Rest connection to devices based on pyATS BaseConnection
+    for DCNM
+
+    YAML Example
+    ------------
+
+        devices:
+            dcnm:
+                connections:
+                    vty:
+                        protocol : ssh
+                        ip : "2.3.4.5"
+                    rest:
+                        class: rest.connector.Rest
+                        ip : "2.3.4.5"
+                        credentials:
+                            rest:
+                                username: admin
+                                password: cisco123
+
+    Code Example
+    ------------
+
+        >>> from pyats.topology import loader
+        >>> testbed = loader.load('/users/xxx/xxx/dcnm123.yaml')
+        >>> device = testbed.devices['dcnm']
+        >>> device.connect(alias='rest', via='rest')
+        >>> device.rest.connected
+        True
+    '''
+
+    @BaseConnection.locked
+    def connect(self, timeout=10):
+        '''connect to the device via REST
+
+        Arguments
+        ---------
+
+            timeout (int): Timeout value
+
+        Raises
+        ------
+
+        Exception
+        ---------
+
+            If the connection did not go well
+
+        Note
+        ----
+
+        There is no return from this method. If something goes wrong, an
+        exception will be raised.
+
+
+        YAML Example
+        ------------
+
+            devices:
+                dcnm:
+                    connections:
+                        vty:
+                            protocol : ssh
+                            ip : "2.3.4.5"
+                        rest:
+                            class: rest.connector.Rest
+                            ip : "2.3.4.5"
+                            credentials:
+                                rest:
+                                    username: admin
+                                    password: cisco123
+
+        Code Example
+        ------------
+
+            >>> from pyats.topology import loader
+            >>> testbed = loader.load('/users/xxx/xxx/dcnm123.yaml')
+            >>> device = testbed.devices['haPrimary']
+            >>> device.connect(alias='rest', via='rest')
+        '''
+
+        self._is_connected = True
```

Comment:
the is connect true should be at the end of the function, in case it blew up somewhere,

Reply:
Addressed